execution/executor-types/src/lib.rs

```diff
@@ -193,6 +193,10 @@ pub trait BlockExecutorTrait: Send + Sync {
         )
     }
 
+    fn pre_commit(&self, block_id: HashValue, parent_block_id: HashValue) -> ExecutorResult<()>;
+
+    fn latest_synced_version(&self) -> Version;
```

Comment:
note: need to distinguish between "synced version" and "pre-committed version" (subject to truncation on restart)